incubator/hnc/internal/kubectl/configset.go

```diff
@@ -37,12 +38,21 @@ var setTypeCmd = &cobra.Command{
 		flags := cmd.Flags()
 		apiVersion, _ := flags.GetString("apiVersion")
 		kind, _ := flags.GetString("kind")
+		force, _ := flags.GetBool("force")
 		config := client.getHNCConfig()
 
 		exist := false
 		for i := 0; i < len(config.Spec.Types); i++ {
 			t := &config.Spec.Types[i]
 			if t.APIVersion == apiVersion && t.Kind == kind {
+				if t.Mode == api.Ignore && mode == api.Propagate && !force {
+					fmt.Println("Switching directly from 'Ignore' to 'Propagate' mode might cause source overwriting conflict.")
+					fmt.Println("If you are sure you want to proceed with this operation, use the '--force' flag.")
+					fmt.Println("If you are not sure and would like to see what source objects would be overwritten," +
+						"please switch to 'Remove' first. Wait 10s before switching it to 'Propagate', which will" +
+						" list all the conflicts that you can resolve them by hand.")
```

Comment:
I'd add a link to https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/docs/user-guide/how-to.md#admin-types, and file a bug to update that topic to discuss this issue.

Also, I'm not sure that 10s is enough? In the doc you can go into more detail; maybe here I'd either say "a minute or two" or just say "to see how to enable propagation safely, see \<link\>."

Reply:
Done. Issue filed at https://github.com/kubernetes-sigs/multi-tenancy/issues/1172